docker/Dockerfile

```diff
@@ -0,0 +1,161 @@
+# dependencies into the production image.
+FROM base-python as builder
+
+# Install any system build dependencies
+COPY docker/build-dependencies.txt /tmp/build-dependencies.txt
+RUN --mount=type=cache,target=/var/cache/apt \
+    /root/docker-apt-install.sh /tmp/build-dependencies.txt
+
+# Install everything in venv for isolation from system python libraries
+RUN python3.10 -m venv /opt/venv
+ENV VIRTUAL_ENV=/opt/venv/ PATH="/opt/venv/bin:$PATH"
+
+# The cache mount means a) /root/.cache is not in the image, and b) it's preserved
+# between docker builds locally, for faster dev rebuild.
+COPY requirements.prod.txt /tmp/requirements.prod.txt
+
+# DL3042: using cache mount instead
+# DL3013: we always want latest pip/setuptools/wheel, at least for now
+# hadolint ignore=DL3042,DL3013
+RUN --mount=type=cache,target=/root/.cache \
+    python -m pip install -U pip setuptools wheel && \
+    python -m pip install --require-hashes --requirement /tmp/requirements.prod.txt
+
+
+##################################################
+#
+# Base project image
+#
+# Ok, we've built everything we need, build an image with all dependencies but
+# no code.
+#
+# Not including the code at this stage has two benefits:
+#
+# 1) this image only rebuilds when the handlful of files needed to build
+# job-server-base changes. If we do `COPY . /app` now, this will rebuild when
+# *any* file changes.
+#
+# 2) Ensures we *have* to mount the volume for dev image, as there's no embedded
+#    version of the code. Otherwise, we could end up accidentally using the
+#    version of the code included when the prod image was built.
+FROM base-python as job-server-base
+
+RUN mkdir -p /app
+WORKDIR /app
+
+# copy venv over from builder image. These will have root:root ownership, but
+# are readable by all.
+COPY --from=builder /opt/venv /opt/venv
+
+# copy node assets over from node-builder image. These will have root:root ownership, but
+# are readable by all.
+COPY --from=node-builder /usr/src/app/assets/dist /opt/assets
+
+ENV VIRTUAL_ENV=/opt/venv/ PATH="/opt/venv/bin:$PATH" BUILT_ASSETS=/opt/assets PYTHONPATH=/app
+
+##################################################
+#
+# Production image
+#
+# Copy code in, add proper metadata
+FROM job-server-base as job-server-prod
+
+# Adjust this metadata to fit project. Note that the base-docker image does set
+# some basic metadata.
+LABEL org.opencontainers.image.title="job-server" \
+      org.opencontainers.image.description="Job Server" \
+      org.opencontainers.image.source="https://github.com/opensafely-core/job-server"
+
+# Expose the port to Dokku
+# https://dokku.com/docs/networking/port-management/#applications-using-expose
+EXPOSE 5000
+
+# copy application code
+COPY . /app
+
+# collect static files
+RUN SECRET_KEY=dummy-key python /app/manage.py collectstatic --no-input
+
+# We set command rather than entrypoint, to make it easier to run different
+# things from the cli
+CMD ["/app/docker/entrypoints/prod.sh"]
```

Comment:
```suggestion
ENTRYPOINT ["/app/docker/entrypoints/prod.sh"]

# We set command rather than entrypoint, to make it easier to run different
# things from the cli
CMD ["gunicorn", "-c", "gunicorn.conf.py", "jobserver.wsgi"]
```